CAVaccineInventory/vial
# Claiming locations

From #657 

<img width="737" alt="image" src="https://user-images.githubusercontent.com/6485006/122155391-bf120480-ce1b-11eb-8f95-d2430e8e9d85.png">

<img width="1128" alt="image" src="https://user-images.githubusercontent.com/6485006/122155547-0c8e7180-ce1c-11eb-9b7c-c956d049ba9c.png">

- [x] add a "claimed" field to the location so users can add themselves as responsible for a location
- [x] and a claimed time/date stamp field as well
- [x] add a filter to the locations table page for Claimed w/ options All / Claimed by you / Claimed by anyone / unclaimed
- [x] show "Is pending review" and "Claimed by" as columns in the locations table page between "request a call" and "full address"
- [x] Add bulk action options "Claim locations" and "Unclaim locations you have claimed"
- [x]  Add Location tags and Location Report Notes 
- [x]  When bulk approving locations, add a note and a tag that says "Approved"
- [x]  Make it so its easier to review claimed locations (review them consecutively)

Additional stuff I did:
- Fix: use logic to display an "s" if the claimed or unclaimed count is greater than 1 in the display message.
- Consolidate logic used by both reporters and locations actions into generic methods

---

How to test this:
- Navigate to the locations list page
- Edit some locations and set `is_pending_review`  flags to true, and on others claim the locations
- Select bulk action options in dropdown on locations list
- Go to the location review note page. 
- Check if a note was created after bulk approving for the location you selected.
- Claim some locations, then go on the locations list page and select the action to mass unclaimed locations. 
- Check `is_pending_review` on several locations, then navigate to a location show page, uncheck `is_pending_review` scroll to the bottom, and select "save and review next pending location". Then verify it worked (correct number of pending review locations, and previous location is not under review.
